mypy/fastparse.py

```diff
@@ -776,7 +776,8 @@ def visit_Str(self, n: ast35.Str) -> Union[UnicodeExpr, StrExpr]:
         def visit_JoinedStr(self, n: ast35.JoinedStr) -> StrExpr:
             result_string_expression = StrExpr('')
             for value in n.values:
-                result_string_expression = OpExpr('+', result_string_expression, self.visit(value))
+                value_as_string_expr = cast(StrExpr, self.visit(value))
+                result_string_expression = cast(StrExpr, OpExpr('+', result_string_expression, value_as_string_expr))
```

Comment:
I used cast because otherwise I thought it would systematically fail without a call to `str(...)`. Am I missing something?

I thought another typesafe solution would be to replace casts by something like (untested) `CallExpr(NameExpr('str'), OpExpr('+', result_string_expression, value_as_string_expr))`?  but it's rather complex compared to untyped casts, and at the end every expression in a f-string is indeed passed to a `str()` call?